### PR TITLE
SOLR-17152: Better alignment of Admin UI graph

### PR DESCRIPTION
**Pull request by janhoy**
https://issues.apache.org/jira/browse/SOLR-17152

### CODE REVIEW

File: solr/webapp/web/js/angular/controllers/cloud.js [hunk 1]
```diff
@@ -971,7 +974,10 @@ solrAdminApp.directive('graph', function(Constants) {
                 var w = element.width(),
                     h = leafCount * 20;
 
-                var tree = d3.layout.tree().size([h, w - 400]);
+                // Calculate roughly the width of host name to align the graph
+                var hostnameWidth = graphData.children.length > 0 ?
+                  graphData.children[0].children[0].children[0].name.length * 7.5 : 400;
```
Comment (epugh) [inline solr/webapp/web/js/angular/controllers/cloud.js:979]: `7.5` must have been some fun trial and error!

File: solr/webapp/web/js/angular/controllers/cloud.js [hunk 2]
```diff
@@ -929,18 +929,21 @@ solrAdminApp.directive('graph', function(Constants) {
...
+                  content: function() {
+                    return $(this).attr('title');
+                  }
                 });
-            });
+              };
+
+            scope.$watch("data", drawGraph);
 
+            window.onresize = drawGraph;
```
Comment (epugh) [inline solr/webapp/web/js/angular/controllers/cloud.js:946]: Nice!  We didn't have this before!

File: solr/webapp/web/js/angular/controllers/cloud.js [hunk 3]
```diff
@@ -971,7 +974,10 @@ solrAdminApp.directive('graph', function(Constants) {
                 var w = element.width(),
                     h = leafCount * 20;
 
-                var tree = d3.layout.tree().size([h, w - 400]);
+                // Calculate roughly the width of host name to align the graph
+                var hostnameWidth = graphData.children.length > 0 ?
+                  graphData.children[0].children[0].children[0].name.length * 7.5 : 400;
+                var tree = d3.layout.tree().size([h, w - hostnameWidth]);
```
Comment (janhoy) [inline solr/webapp/web/js/angular/controllers/cloud.js]: I have not tested with different host names.
Guess the perfect formula would be `w - legendWidth - hostnameWidth`

File: solr/webapp/web/js/angular/controllers/cloud.js (hunk 1 above)
Comment (janhoy) [reply solr/webapp/web/js/angular/controllers/cloud.js]: Will test with a few other widths I suppose. I'm trying a formula with substracting legend box width and then calculating hostname width using chars * `5.5` :)

**Review by risdenk — APPROVED** [review]

File: solr/webapp/web/js/angular/controllers/cloud.js [hunk 4]
```diff
@@ -971,7 +974,10 @@ solrAdminApp.directive('graph', function(Constants) {
                 var w = element.width(),
                     h = leafCount * 20;
 
-                var tree = d3.layout.tree().size([h, w - 400]);
+                // Calculate roughly the width of host name to align the graph
+                var hostnameWidth = graphData.children.length > 0 ?
+                  graphData.children[0].children[0].children[0].name.length * 5.5 : 400;
+                var tree = d3.layout.tree().size([h, w - 140 - hostnameWidth]);
```
Comment (janhoy) [inline solr/webapp/web/js/angular/controllers/cloud.js:980]: Perhaps also make sure `w - 140 - hostnameWidth` has a certain minimum width, such as 200?

**Pull request merged**